.github/workflows/ci.yml

```diff
@@ -22,7 +31,18 @@ jobs:
         toolchain: ${{ matrix.rust }}
         profile: minimal
         override: true
+        components: rustfmt
 
     - name: Run tests
       run: |
         cargo test
+
+    - name: Run tests using no_std
+      if: matrix.test_no_std == true
+      run: |
+        cargo test --no-default-features --features alloc
+
+    - name: Check formatting
+      if: matrix.rust == 'stable'
+      run: |
+        cargo fmt --all -- --check
```

Comment:
Note to myself: I'll move it to another CI check, along with clippy probably

Reply:
`ripgrep` actually uses a separate job: https://github.com/BurntSushi/ripgrep/blob/a2e6aec7a4d9382941932245e8854f0ae5703a5e/.github/workflows/ci.yml#L185-L200

Reply:
Yep, I have something like that for the Rust repos at work: https://github.com/onecodex/needletail/blob/master/.github/workflows/tests.yml